edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/TwinManager.cs

```diff
@@ -265,21 +265,26 @@ static void ValidateTwinCollectionSize(TwinCollection collection)
 
         static void ValidateTwinProperties(JToken properties, int currentDepth)
         {
-            foreach (JProperty kvp in ((JObject)properties).Properties())
+            foreach (JProperty kvp in ((JObject) properties).Properties())
             {
                 ValidatePropertyNameAndLength(kvp.Name);
 
                 ValidateValueType(kvp.Name, kvp.Value);
 
-                string s = kvp.Value.ToString();
-                ValidatePropertyValueLength(kvp.Name, s);
-
-                if ((kvp.Value is JValue) && (kvp.Value.Type is JTokenType.Integer))
+                if (kvp.Value is JValue)
                 {
-                    ValidateIntegerValue(kvp.Name, (long)kvp.Value);
+                    if (kvp.Value.Type is JTokenType.Integer)
```

Comment:
How many different types are their? Is it only integer and string? Where do we reject arrays?

Reply:
Arrays get rejected at line 272.

Reply:
For types other than integer, it will be treated as string (that what we have previously).  You can find all supported types from JsonEx.ValidDiffTypes.